Objects/Converters/ConverterRhinoGh/ConverterRhinoGhShared/ConverterRhinoGh.Other.cs

```diff
@@ -66,36 +67,71 @@ public Rhino.Render.RenderMaterial RenderMaterialToNative(RenderMaterial speckle
+//#if RHINO6
       var rhinoMaterial = new Material
       {
         Name = speckleName,
         DiffuseColor = Color.FromArgb(speckleMaterial.diffuse),
         EmissionColor = Color.FromArgb(speckleMaterial.emissive),
-        Transparency = 1 - speckleMaterial.opacity,
-        Reflectivity = speckleMaterial.metalness
+        Transparency = 1 - speckleMaterial.opacity
       };
+      rm = Rhino.Render.RenderMaterial.CreateBasicMaterial(rhinoMaterial, Doc);
+//#else
+      //TODO Convert materials as PhysicallyBasedMaterial 
+      // var pbrRenderMaterial = RenderContentType.NewContentFromTypeId(ContentUuids.PhysicallyBasedMaterialType, Doc) as Rhino.Render.RenderMaterial;
+      // RH.Material simulatedMaterial = pbrRenderMaterial.SimulatedMaterial(RenderTexture.TextureGeneration.Allow);
+      // RH.PhysicallyBasedMaterial pbr = simulatedMaterial.PhysicallyBased;
+      //
+      // pbr.BaseColor = ARBGToColor4f(speckleMaterial.diffuse);
+      // pbr.Emission = ARBGToColor4f(speckleMaterial.emissive);
+      // pbr.Opacity = speckleMaterial.opacity;
+      // pbr.Metallic = speckleMaterial.metalness;
+      // pbr.Roughness = speckleMaterial.roughness;
+      //
+      // rm = Rhino.Render.RenderMaterial.FromMaterial(pbr.Material, Doc);
+      // rm.Name = speckleName;
+//#endif
       
-      var renderMaterial = Rhino.Render.RenderMaterial.CreateBasicMaterial(rhinoMaterial, Doc);
-      Doc.RenderMaterials.Add(renderMaterial);
+      Doc.RenderMaterials.Add(rm);
 
-      return renderMaterial;
+      return rm;
     }
     public RenderMaterial RenderMaterialToSpeckle(Material material)
     {
       var renderMaterial = new RenderMaterial();
       if (material == null) return renderMaterial;
 
-      renderMaterial.name = (material.Name == null) ? "default" : material.Name; // default rhino material has no name or id
+      renderMaterial.name = material.Name ?? "default"; // default rhino material has no name or id
+#if RHINO6
+      
       renderMaterial.diffuse = material.DiffuseColor.ToArgb();
       renderMaterial.emissive = material.EmissionColor.ToArgb();
       renderMaterial.opacity = 1 - material.Transparency;
-      renderMaterial.metalness = material.Reflectivity;
-
+      
       // for some reason some default material transparency props are 1 when they shouldn't be - use this hack for now
       if ((renderMaterial.name.ToLower().Contains("glass") || renderMaterial.name.ToLower().Contains("gem")) && renderMaterial.opacity == 0)
         renderMaterial.opacity = 0.3;
-
+#else
+      Material matToUse = material;
+      if(!material.IsPhysicallyBased)
+      {
+        matToUse = new Material();
+        matToUse.CopyFrom(material);
+        matToUse.ToPhysicallyBased();
+      }
+      using (var rm = Rhino.Render.RenderMaterial.FromMaterial(matToUse, null))
+      {
+        RH.PhysicallyBasedMaterial pbrMaterial = rm.ConvertToPhysicallyBased(RenderTexture.TextureGeneration.Allow);
```

Comment:
Did you check if this conversion to a physically based material returns the correct opacity? Just in case this means we can get rid of the transparency hack in line 114

Reply:
From Rhino 7, all properties including opacity are correct. I've just added screenshots to my PR. 

That hack is only necessary for Rhino 6, which doesn't support any of the Physically Based functions we are using here.